### ISSUE TITLE
SAPI5 (x64): NVDA fails to get voice info for some 64-bit voices (eg Divvun Saami)

### ISSUE THREAD

**Issue by zstanecic** [nvaccess/nvda#19060]
### Brief summary

The synthesizer from 
https://pahkat.uit.no/divvun-installer/download/divvun-installer?platform=windows
which started working on sapi5 on the x64 builds of NVDA is not properly initialized

### Steps to reproduce

1. Download the divvun manager from: https://pahkat.uit.no/divvun-installer/download/divvun-installer?platform=windows
2. When installed, download any of the southern or northern saami voices
3. try to switch to sapi5 and select the downloaded voice from the voice settings.

### Actual behavior

The wrong voice name is reported, instead of the actual voice name, and NVDA cannot get the info about the voice selected.

### Expected behavior

NVDA should properly report selected voice. 

### NVDA logs, crash dumps and other attachments

The log file states the following:
WARNING - synthDrivers.sapi5.SynthDriver._getAvailableVoices (15:44:13.299) - MainThread (1500):
Could not get the voice info. Skipping...
Registry path where the voice is registered in the registry is: Komputer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices\Tokens\DivvunTTS_se_biret

### NVDA type

installed copy

### NVDA version

alpha-52970,c9b9d021 (2026.1.0.52970)

### Have you tried any other versions of NVDA? If so, please report their behaviors.

This is applicable only to x64 versions of NVDA, because x64 versions of NVDA introduced x64 sapi5 support.

### Windows version

Windows 11 25H2 (10.0.26200.6725) workstation AMD64

### Name and version of other software in use when reproducing the issue

Not applicable.

### Other information about your system

Tested on a fresh installation of windows.

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by gerald-hartig** [comment]
@zstanecic This looks like an installation issue with your voice. Can you please try to fully uninstall the voice and try again?

**Comment by snomos** [comment]
> [@zstanecic](https://github.com/zstanecic) This looks like an installation issue with your voice. Can you please try to fully uninstall the voice and try again?

What kind of installation issue could it be? Maybe @bbqsrc can help with this if it is on the Divvun side.

**Comment by zstanecic** [comment]
No, the same happens in jaws. When you select the voice with arrows, they don't speak. Only when trying to speak individual words, plus these are not responsive for a daily use.

**Comment by zstanecic** [comment]
and yeah, the voices are dislpayed like they pretend to be Microsoft's voices.

**Comment by seanbudd** [comment]
as discussed in https://github.com/nvaccess/nvda/issues/17661#issuecomment-3546235668, the voices have metadata issues that need to be fixed by developers

**Comment by bbqsrc** [comment]
@seanbudd anything more specific? https://github.com/nvaccess/nvda/blob/36a2e7d4f6b83ffa5f4c3003fb5892a5f516de6f/source/synthDrivers/sapi5.py#L619-L635 doesn't tell me much, since the actual COM error is swallowed.

**Comment by seanbudd** [comment]
@bbqsrc - you may want to create a PR or local build which logs the information you need

**Comment by gexgd0419** [comment]
The above code fails on the line `name = v[i].GetDescription()`, because the specific voice doesn't have a "language independent name", as said in Microsoft's [Object Tokens and Registry Settings White Paper](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/ms717036(v=vs.85)):

| RegKey | ValueName | Sample Value | Comments |
| -- | -- | -- | -- |
| SampleTokenKey |   |   | Required - This is the RegKey for the Token. |
|  | **(Default)** | Joe | **Required - Language Independent Name.** |
|  | 409 | Joe | Name in Hex LangID 409, which is English. There may be several of these rows, one for each LangID in which the Token has a name. Note, no leading 0x before the LangID. |
|  | 809 | Joe |
|  | CLSID | {8021D50E-D93C-4075-8504-FC4E124D64E9} | Required - Sample CLSID for object which instantiates the token. |

The language independent name, which is the default (unnamed) string value of the voice registry key, is **required**, according to the SAPI5 spec. That voice, however, only has a `409` string value, which is the English (US) name of the voice, with the default value left unset.

So if the OS language is anything but English (US), `GetDescription()` will fall back to the language independent name and, since there is none, fail with `SPERR_NOT_FOUND` (`0x8004503a`).

The fix for the voice is simple: copy the English name as the language independent name, that is, the unnamed value of the voice registry key.

Narrator doesn't fail, because it tries to read the `Name` attribute first, then fall back to `GetDescription`. And that voice does have a proper `Name` attribute, so Narrator will happily accept it, but NVDA, which only tries `GetDescription`, fails.

Note that `GetDescription` usually gets a longer, more descriptive name, such as `Microsoft Zira Desktop - English (United States)`, while the `Name` attribute is usually shorter, such as `Microsoft Zira Desktop`. The attribute filter only works on the latter "name".
